node/node.go

```diff
@@ -1124,7 +1124,7 @@ func (node *Node) handleSubscription(ctx context.Context, f pubSubProcessorFunc,
 			return
 		}
 
-		if err := f(ctx, pubSubMsg); err != nil {
+		if err := f(ctx, pubSubMsg); err != nil && err != context.Canceled {
```

Comment:
There is a lot going on in this line. Consider breaking out the cancelled check to another if statement.